packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -775,6 +742,43 @@ export function clearContainer(container: Container): void {
   }
 }
 
+function clearContainerChildren(container: Node) {
+  let node;
+  let nextNode: ?Node = container.firstChild;
+  while (nextNode) {
+    node = nextNode;
+    nextNode = nextNode.nextSibling;
+    switch (node.nodeName) {
+      case 'HTML':
+      case 'HEAD':
+      case 'BODY': {
+        const element: Element = (node: any);
+        clearContainerChildren(element);
```

Comment:
Any reason this couldn't just call clearContainer recursively instead of as a separate function?

Reply:
To avoid a type check when called recursively. If we just called clearContainer we would need to recheck if the target is a Document or Element